test-site/src/components/MapPin.tsx

```diff
@@ -1,18 +1,55 @@
-import { PinComponent } from '@yext/search-ui-react';
+import { useEffect, useRef, useState } from 'react';
+import { Map, Popup, LngLatLike } from 'mapbox-gl';
+import { PinComponent, Coordinate } from '@yext/search-ui-react';
+import { Result } from '@yext/search-headless-react';
 import { Location } from '../pages/LocationsPage';
 
-export const MapPin: PinComponent<Location> = () => {
+export interface MapPinProps<T> {
+  coordinate: Coordinate,
+  index: number,
+  mapbox: Map,
+  result: Result<T>
+}
+
+const transformToMapboxCoord = (coordinate: Coordinate): LngLatLike => ({
+  lng: coordinate.longitude,
+  lat: coordinate.latitude,
+});
+
+export const MapPin: PinComponent<Location> = (props: MapPinProps<Location>) => {
+  const { coordinate, mapbox, result } = props;
+  const [active, setActive] = useState(false);
+  const popupRef = useRef(new Popup({ offset: 15 })
+    .on('close', () => setActive(false))
+  );
+
+  useEffect(() => {
+    if (active) {
+      const popup = document.createElement('div');
+      popup.innerHTML = `<div>${result.name}</div>`;
```

Comment:
I double checked in browser console and here popup will be a div with a div inside it `<div><div>result.name</div></div>`, when I think we just want `<div> result.name </div>`. Maybe just `popup.innerText = result.name`?